src/common/transformations/src/transformations/common_optimizations/gru_cell_fusion.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/gru_cell_fusion.hpp"
+
+#include <memory>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace std;
+using namespace ov::opset9;
+using namespace ov::pass::pattern;
+
+ov::pass::GRUCellFusion::GRUCellFusion() {
+    MATCHER_SCOPE(GRUCellFusion);
+
+    auto concat_1 = wrap_type<Concat>({any_input(), any_input()});
```

Comment:
Yes, these transformations are very similar, but AUGRUCell op has a really limited support now, e.g. only {Sigmoid, Tanh} activations are supported, the pattern has strict structure with Biases. And I'm still not sure about the place where we will call AUGRUFusion Transformation (before or after CommonTransformations pipeline), the pattern can be different depends on the place. So that's why I decided to have AUGRUCell fusion as a separate transformation, it helps as to quickly change this transformation.
When AUGRUCell op becomes part of the official opset, we will be able to partially combine these transformations.

About the lambda, I hope it should work, I will try to apply it

Reply:
done